vllm/v1/core/sched/scheduler.py

```diff
@@ -806,6 +819,79 @@ def _update_after_schedule(
         # it will also affect the scheduler output.
         self.finished_req_ids = set()
 
+    def _update_request_as_session(self, session: Request) -> None:
+        """
+        Updates the waiting session with the next streaming request.
+
+        Removes the last output token (not yet scheduled) from `_all_token_ids`
+        because the new request's prompt tokens will replace it. Typically the decoded
+        outputs are scheduled as the next input in autoregressive decoding. When we
+        receive a new streaming request, the new prompt becomes our next input, so the
+        last output token is no longer needed and will not join the kv cache. This
+        ensures correct calculation of `num_new_tokens` in `schedule`.
+        """
+        assert session.streaming_queue is not None
+        request = session.streaming_queue.popleft()
+
+        num_new_tokens = session.num_tokens - session.num_computed_tokens
+        assert num_new_tokens in {0, 1}, f"got {num_new_tokens=}"
```

Comment:
nit: tuple probably better than set here
```suggestion
        assert num_new_tokens in (0, 1), f"got {num_new_tokens=}"
```